command/genesis/genesis.go

```diff
@@ -164,23 +190,48 @@ func setLegacyFlags(cmd *cobra.Command) {
 	_ = cmd.Flags().MarkHidden(chainIDFlagLEGACY)
 }
 
-func runPreRun(_ *cobra.Command, _ []string) error {
+func runPreRun(cmd *cobra.Command, _ []string) error {
 	if err := params.validateFlags(); err != nil {
 		return err
 	}
 
+	helper.SetRequiredFlags(cmd, params.getRequiredFlags())
+
 	return params.initRawParams()
 }
 
 func runCommand(cmd *cobra.Command, _ []string) {
 	outputter := command.InitializeOutputter(cmd)
 	defer outputter.WriteOutput()
 
-	if err := params.generateGenesis(); err != nil {
-		outputter.SetError(err)
+	var result command.CommandResult
+
+	if params.isPolyBFTConsensus() {
+		config, err := params.GetChainConfig()
+		if err != nil {
+			outputter.SetError(err)
+
+			return
+		}
+
+		if err := helper.WriteGenesisConfigToDisk(config, params.genesisPath); err != nil {
+			outputter.SetError(err)
+
+			return
+		}
+
+		result = &GenesisResult{
+			Message: fmt.Sprintf("Genesis written to %s\n", params.genesisPath),
+		}
```

Comment:
Why not extract this out, in a similar method to `generateGenesis`?

This way we avoid this if / else spaghetti

Reply:
done

Reply:
Fixed where?